Gordon360/Services/ProfileService.cs

```diff
@@ -274,6 +276,111 @@ public async Task UpdateCustomProfileAsync(string username, string type, CUSTOM_
+        {
+            if (row.Visibility == "Private" || (row.Visibility == "FacStaff" && currentUserType != "fac"))
+            {
+                fs_vm.GetProperty(row.Field).SetValue(publicFac, "Private as requested.");
+            }
+        }
+
+        return publicFac;
+    }
+
+    /// <summary>
+    /// convert original student profile to public student profile based on individual privacy settings
+    /// </summary>
+    /// <param name="username">username of the student being searched</param>
+    /// <param name="currentUserType">personnel type of the logged-in user</param>
+    /// <param name="stu">original profile of the student being searched</param>
+    /// <returns>public profile of the student based on individual privacy settings</returns>
+    public PublicStudentProfileViewModel ToPublicStudentProfileViewModel
+        (string username, string currentUserType, StudentProfileViewModel stu)
+    {
+        PublicStudentProfileViewModel publicStu = (PublicStudentProfileViewModel)stu;
+        var account = accountService.GetAccountByUsername(username);
+
+        // select all privacy settings
+        var privacy = context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID);
+        // get type of viewmodel for reflection property setting
+        Type s_vm = new PublicStudentProfileViewModel().GetType();
+
+        foreach (UserPrivacy_Settings row in privacy)
+        {
+            if (row.Visibility == "Private" || (row.Visibility == "FacStaff" && currentUserType != "fac"))
+            {
+                s_vm.GetProperty(row.Field).SetValue(publicStu, "Private as requested.");
+            }
+        }
+
+        return publicStu;
+    }
+
+    /// <summary>
+    /// get privacy setting for particular user
+    /// </summary>
+    /// <param name="username">AD username</param>
+    /// <returns>all privacy setting of the particular user</returns>
+    public IEnumerable<UserPrivacyViewModel> GetPrivacySettingAsync(string username)
+    {
+        var account = accountService.GetAccountByUsername(username);
+
+        // select all privacy settings
+        var privacy = context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID).Select(up_s => (UserPrivacyViewModel)up_s);
+
+        return privacy;
+    }
+
+    /// <summary>
+    /// privacy setting of some piece of personal data for user.
+    /// </summary>
+    /// <param name="username">AD Username</param>
+    /// <param name="facultyStaffPrivacy">Faculty Staff Privacy View Model</param>
+    public async Task UpdateUserPrivacyAsync(string username, UserPrivacyUpdateViewModel facultyStaffPrivacy)
+    {
+        var account = accountService.GetAccountByUsername(username);
+        foreach (string subField in facultyStaffPrivacy.Field)
+        {
+            var facStaff = context.UserPrivacy_Settings.FirstOrDefault(x => x.gordon_id == account.GordonID && x.Field == subField);
```

Comment:
Is there a reason that you changed how you were abbreviating for this one? You used `up_s` above, but only have `x` here. I don't mind either, but it's best to be consistent where possible.